src/vm/eventpipe.cpp

```diff
@@ -339,31 +290,53 @@ EventPipeSessionID EventPipe::Enable(
     }
     CONTRACTL_END;
 
-    // If tracing is not initialized or is already enabled, bail here.
-    if (!s_tracingInitialized || s_pConfig == nullptr || s_pConfig->Enabled())
+    // If the state or arguments are invalid, bail here.
+    if (pSession == nullptr || !pSession->IsValid())
+        return 0;
+    if (sessionType == EventPipeSessionType::File && strOutputPath == nullptr)
```

Comment:
It depends whether violating these invariants represent a bug in the runtime or unexpected behavior from an external components, ie a client communicating via IPC or user-written managed code. Presumably our interface will define rules such as "requesting a stream written to file requires a non-null path". If an IPC message include a NULL path that needs to be validated somewhere, and if that validation is right here then its reasonable for it not to be a precondition. On the other hand if the runtime should have guaranteed the condition is true already then it should be a precondition. For example pSession != nullptr should be probably be a precondition because the runtime caller created that pointer.

Reply:
I am hoping the change can check the invalid data as soon as they arrive and stop the call fast. The earlier we do the check, the less code needs to potentially check against invalid data, the less risk for us to have bugs.

Reply:
In general I'd recommend putting the validation in the earliest place that allows it be shared across all incoming call paths. If there are significant side-effects that occur before the paths merge (hopefully there aren't) then I agree it would make sense to clone the validation and move it up towards individual edge entrypoints.

That said, personally I'd fine merging a PR that did not match my recommendation as long as it wasn't egregous or security sensitive and the tests aren't finding issues in the input handling overall.